interview-questions.md

```diff
@@ -8,62 +8,105 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 3. What is the difference between console.log() and return?
 
   Your answer:
+  console.log is a way in javascript to see what is happening to my code in the terminal. I can see how my data are changing and have a reference of what I'm doing or simply get a message when I run my code.
+  Return is associated to functions and it's part of its syntax, I can have only one return in a function but there's no limimtations for control.log
 
   Researched answer:
 
+  console.log() can be called as many times as we want and we can even  inside functions to see what is my output in that specific point
+  Return is part of a function syntax and it's needed to call the function, following the instructions to perform a task but itself it will not give an output in the terminal, to see the output I need to console log the function.
 
 
 4. In regards to functions, what is an argument?
 
   Your answer:
 
+ argument is the data the function is acting on.
+
   Researched answer:
 
+  Argument is the external information that a function is accessing and it's often requred to run.
+  
+  Notice that in the function we created a placeholder called name. This allows us to pass any "data" we want through the function during the function call. the "data" can be a placeholder or can contains other information like variables.  
+ 
+
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
   Your answer:
 
+ In pair programing you have two people working on the same code, the two share the rationale and one writes the code and the second person double checks if it's well written, there are no typos and gives feedback and suggestions. It's a way to mutually help in case of mistakes and doubts.
+
   Researched answer:
 
+Pair programming consiste of two people the navigator and the driver working on the same code. In pair programming the driver is responsible for entering the code and follow the instructions given by the navigator. the navigator dictate the code and is responsible to check typing errors. proper communication is key as much as balance out the contribution of each person, hold each other accountable to obtain a better code in less time, the typying syntax and problem solving are split between two people making the overall process easier. 
 
 
 6. What is TDD? Describe the work flow associated with TDD.
 
   Your answer:
+ TDD is a way to check that the code is working and acting as expected.
+
+ the first step is to describe what my function will do, what it is testing and what is the return.
+ the syntax is 
+ describe: function tested
+ "test: describe with your own words what's the scope of your code
+ expect: name of the function
+ toEqual: output of the function tested
+
+ Followed by the actual code with the function tested
+ running in the terminal my js file I will get a message (red/green) describing if test was passed or failed and give indications of the issue
```

Comment:
Excellent! I think it would be wise to also include that "TDD" stands for "Test Driven Development"